training/distributed_training/Dockerfile

```diff
@@ -0,0 +1,101 @@
+FROM nvidia/cuda:9.0-devel-ubuntu16.04
+
+# TensorFlow version is tightly coupled to CUDA and cuDNN so it should be selected carefully
+ENV TENSORFLOW_VERSION=1.10.0
+ENV CUDNN_VERSION=7.0.5.15-1+cuda9.0
+ENV NCCL_VERSION=2.2.13-1+cuda9.0
+
+
+
+RUN apt-get update && apt-get install -y --no-install-recommends --allow-downgrades --allow-change-held-packages \
+      build-essential \
+      cmake \
+      git \
+      curl \
+      vim \
+      wget \
+      ca-certificates \
+      libcudnn7=${CUDNN_VERSION} \
+      libnccl2=${NCCL_VERSION} \
+      libnccl-dev=${NCCL_VERSION} \
+      libjpeg-dev \
+      libpng-dev  \
+      openssl \
+      libssl-dev \
+      unzip \
+      net-tools
+
+# Install Python3
+RUN wget https://www.python.org/ftp/python/3.6.6/Python-3.6.6.tgz
+RUN tar -xvf Python-3.6.6.tgz
+
+RUN cd Python-3.6.6 && ./configure && make && make install
+
+
+RUN echo "deb http://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1604/x86_64 /" > /etc/apt/sources.list.d/nvidia-ml.list
+
+
+RUN ln -s /usr/local/bin/python3.6 /usr/bin/python
+RUN ln -s /usr/local/bin/pip3 /usr/bin/pip
+
+RUN apt-get install -y --no-install-recommends --allow-downgrades --allow-change-held-packages libgtk2.0-dev
+
+RUN pip3 install --upgrade pip
+RUN pip3 install awscli
+
+
+# Install Open MPI
+RUN mkdir /tmp/openmpi && \
+    cd /tmp/openmpi && \
+    wget https://www.open-mpi.org/software/ompi/v3.0/downloads/openmpi-3.0.0.tar.gz && \
+    tar zxf openmpi-3.0.0.tar.gz && \
+    cd openmpi-3.0.0 && \
+    ./configure --enable-orterun-prefix-by-default && \
+    make -j $(nproc) all && \
+    make install && \
+    ldconfig && \
+    rm -rf /tmp/openmpi
+
+
+# Install TensorFlow and Keras
+RUN pip3 install tensorflow-gpu==${TENSORFLOW_VERSION} keras h5py
+
+# Install Horovod, temporarily using CUDA stubs
+RUN ldconfig /usr/local/cuda-9.0/targets/x86_64-linux/lib/stubs && \
+    HOROVOD_GPU_ALLREDUCE=NCCL HOROVOD_WITH_TENSORFLOW=1 pip3 install --no-cache-dir horovod==0.13.11 && \
```

Comment:
Can we use the most recent version of Horovod?

Reply:
sure , will fix in next iteration